frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/utils/optionsUtils.test.ts

```diff
@@ -1,120 +1,238 @@
+  describe('getSelectedOptionsTypeWithManualSupport', () => {
+    it('should return SelectedOptionsType.Unknown if both options and optionsId are set', () => {
+      const codeListId = 'codeListId';
+      const options: Option[] = [{ label: 'label1', value: 'value1' }];
+      const optionListIds = ['codeListId'];
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
+      expect(result).toEqual(SelectedOptionsType.Unknown);
+    });
+
+    it('should return SelectedOptionsType.CodeList if options is not set and codeListId is in optionListIds', () => {
+      const codeListId = 'codeListId';
+      const options = undefined;
+      const optionListIds = ['codeListId'];
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
+      expect(result).toEqual('codelist');
+    });
+
+    it('should return SelectedOptionsType.ReferenceId if options is not set and codeListId is not in optionListIds', () => {
+      const codeListId = 'codeListId';
+      const options = undefined;
+      const optionListIds = ['anotherCodeListId'];
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
+      expect(result).toEqual(SelectedOptionsType.ReferenceId);
+    });
+
+    it('should return SelectedOptionsType.Manual if options is set and codeListId is not set', () => {
+      const codeListId = undefined;
+      const options = [{ label: 'label1', value: 'value1' }];
+      const optionListIds = ['anotherCodeListId'];
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
+      expect(result).toEqual(SelectedOptionsType.Manual);
+    });
+
+    it('should return SelectedOptionsType.Manual if options is set and codeListId is not set, even if options has length 0', () => {
+      const codeListId = undefined;
+      const options = [];
+      const optionListIds = ['anotherCodeListId'];
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
+      expect(result).toEqual(SelectedOptionsType.Manual);
+    });
+
+    it('should use default value for optionListIds if it is not provided', () => {
+      const codeListId = '';
+      const options = undefined;
+      const result = getSelectedOptionsTypeWithManualSupport(codeListId, options);
+      expect(result).toEqual(SelectedOptionsType.CodeList);
+    });
   });
 
-  it('should return SelectedOptionsType.CodeList if options is not set and codeListId is in optionListIds', () => {
-    const codeListId = 'codeListId';
-    const options = undefined;
-    const optionListIds = ['codeListId'];
-    const result = getSelectedOptionsTypeWithManualSupport(codeListId, options, optionListIds);
-    expect(result).toEqual('codelist');
+  describe('componentUsesDynamicCodeList', () => {
+    it('should return false if codeListId is not set', () => {
```

Comment:
I know we are going to change this permission, and therefore also the test, in the future, but it is more correct to phrase it like this:
```suggestion
    it('should return false if codeListId is set to an empty string', () => {
```